### PR TITLE
Remove trailing slash on plist path

### LINKED ISSUE #5018: appium is crashing on startup. again
The logs are [in this gist](https://gist.github.com/bootstraponline/bfaee92a8796d54123c2). I'm running the latest appium master rev and I double checked that the submodules are up to date. 

`error: Could not determine Xcode version:Could not get Xcode version. /Applications/Xcode.app/Contents/Info.plist/ does not exist on disk.`

I'm using Xcode 6.3.1 on OS X 10.10.3. I can't make a new release of the Ruby bindings because the server crashes on startup which makes executing the iOS test suite difficult.

### PR DESCRIPTION
**Pull request by bootstraponline**
Fix https://github.com/appium/appium/issues/5018

### CODE REVIEW

File: lib/xcode.js [hunk 1]
```diff
@@ -109,7 +109,8 @@ async function getVersionWithoutRetry () {
 
   // we want to read the CFBundleShortVersionString from Xcode's plist.
   // It should be in /[root]/XCode.app/Contents/
-  const plistPath = xcodePath.replace(XCODE_SUBDIR, "/Contents/Info.plist");
+  let plistPath = xcodePath.replace(XCODE_SUBDIR, "/Contents/Info.plist");
+  plistPath = plistPath.replace(new RegExp("/$"), "").trim();
```
Comment (jlipps) [inline lib/xcode.js]: before I merge this I'd like to understand how `plistPath` could ever have a trailing slash at this point in the code. I thought that `getPath()` would always return a string without a trailing slash based on previous commits

Comment (bootstraponline) [inline lib/xcode.js]: I'd like you to execute the following:

```
var xcodePath = '/Applications/Xcode.app/Contents/Developer/';
var plistPath = xcodePath.replace("/Contents/Developer",  "/Contents/Info.plist");
```

then tell me what the value of plistPath is.

Comment (sebv) [inline lib/xcode.js]: The standard way to do that in node is to use `path.resolve` everywhere.
https://nodejs.org/api/path.html#path_path_resolve_from_to

Comment (bootstraponline) [inline lib/xcode.js]: `plistPath.replace(new RegExp("/$"), "").trim();` is a direct copy and paste from existing code in appium-xcode. maybe the entire file should be rewritten to use resolve however it's out of scope for this bug fix.

Comment (sebv) [inline lib/xcode.js]: Definitely we need to rewrite that.

But just in this line you are reinventing the wheel, you can just do:

```
path.join(plistPath, '.')
```

Edit: can try gist here: https://gist.github.com/sebv/960e372891e156903976

**Comment by bootstraponline** [comment]
Ok, I've updated to use resolve as requested.

**Comment by jlipps** [comment]
ok, this makes more sense. thanks

**Pull request merged**